clang/lib/Index/IndexUnitWriter.cpp

```diff
@@ -89,8 +104,16 @@ class IndexUnitWriter::PathStorage {
         while (!dirStr.empty() && dirStr[0] == '/')
           dirStr = dirStr.drop_front();
       }
-      dirPath.Dir.Offset = getPathOffset(dirStr);
-      dirPath.Dir.Size = dirStr.size();
+
+      if (dirPath.PrefixKind != UNIT_PATH_PREFIX_NONE) {
```

Comment:
Yeah, we could, at the downside of needing a version bump, plus some slight duplication since most things are relative to the workdir/sysroot in practice.

Reply:
I think I convinced myself it's wrong anyway. I don't think the reader should know whether it was mapped or not in the prefix map case.